### DISCUSSION TITLE
[Feature request] Nested builder with drag-and-drop

### DISCUSSION THREAD

**Discussion by binaryfire** [filamentphp/filament#11838] [Ideas]
Feature request for the builder field: drag-and-drop nesting (or a "group" block that allows dragging blocks in / out of it). It's very common to have to group content when you're building things visually. And a repeater doesn't work for this because you can't drag things into / out of a repeater (or between repeaters).

I don't have the JS chops to PR this myself but I think it'd be pretty easy for a JS / Sortable guru.  It'd open up so many more possibilities for the builder field.

![image](https://github.com/filamentphp/filament/assets/5361908/7ddbc5a7-848a-48f4-a569-1cfe8973fafe)

**Comment by Max-Lovell** [comment]
+1 - I'm trying to add if conditions and loops to zapier-like automation system and being able to drag and drop between multiple nested layers would be such a nice UI feature

**Comment by HoceineEl** [comment]
Not sure if you still need this @binaryfire, but I made a page builder for Filament (v4/v5) that works visually: sections in a live preview, and a section builder where you can nest blocks inside containers, columns, cards etc. by drag and drop.

Demo if you want to try it: https://demo.filamentcraft.dev/launch/admin

**Comment by binaryfire** [comment]
Closing this as I’m not using Laravel right now. If anyone still needs this, please re-open a new discussion.
